src/cargo/core/compiler/job_queue.rs

```diff
@@ -266,6 +265,7 @@ impl<'a> JobQueue<'a> {
         // profiles used.  However, to keep it simple and compatible with old
         // behavior, we just display what the base profile is.
         let profile = cx.bcx.profiles.base_profile(self.is_release);
```

Comment:
Hm, I *think* the comment here is intended to refer primarily to `optimized` and `debuginfo` settings, because different crates might be compiled with different optimization/debuginfo settings, even if they are linked into a single artifact at the end. Nevertheless, the profile is the same across the whole compilation, the variation is internal to profile.

We also are thinking about removing profiles other than dev/release, so I would say that the current behavior is what we actually want, even if it is not 100% technically accurate.

Reply:
OK. No harm done.

Perhaps some of what you said should appended to #4140.